lib/srv/db/ca.go

```diff
@@ -103,16 +104,30 @@ func (s *Server) getCACert(ctx context.Context, database types.Database) ([]byte
 // getCACertPath returns the path where automatically downloaded root certificate
 // for the provided database is stored in the filesystem.
 func (s *Server) getCACertPath(database types.Database) (string, error) {
-	// All RDS and Redshift instances share the same root CA which can be
-	// downloaded from a well-known URL (sometimes region-specific). Each
-	// Cloud SQL instance has its own CA.
 	switch database.GetType() {
+	// All RDS instances share the same root CA (per AWS region) which can be
+	// downloaded from a well-known URL.
 	case types.DatabaseTypeRDS:
 		return filepath.Join(s.cfg.DataDir, filepath.Base(rdsCAURLForDatabase(database))), nil
+
+	// All Redshift instances share the same root CA which can be downloaded
+	// from a well-known URL.
 	case types.DatabaseTypeRedshift:
 		return filepath.Join(s.cfg.DataDir, filepath.Base(redshiftCAURLForDatabase(database))), nil
+
+	// ElastiCache databases are signed with Amazon root CA. In most cases,
+	// x509.SystemCertPool should be sufficient to verify ElastiCache servers.
+	// However, x509.SystemCertPool does not support windows for go versions
```

Comment:
I don't mind the logic, but I don't think that we support Windows anyways for the servers nor AWS (I can be wrong here??). I also think that we should consider updating Go for Teleport 10.

Reply:
hmm I didn't know if we support windows servers or not. I would really like Go upgraded. Want to start the generics stuff.